sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -588,6 +589,27 @@ private[hive] class HiveMetastoreCatalog(val client: ClientInterface, hive: Hive
       // Wait until children are resolved.
       case p: LogicalPlan if !p.childrenResolved => p
       case p: LogicalPlan if p.resolved => p
+
+      case CreateViewAsSelect(table, child, allowExisting, orReplace, sql) =>
+        if (conf.canonicalizeView) {
+          if (allowExisting && orReplace) {
+            throw new AnalysisException("Can't combine IF NOT EXISTS and OR REPLACE.")
```

Comment:
Maybe `It is not allowed to define a view with both IF NOT EXISTS and OR REPLACE.`?